### PR TITLE
Contest - edit contest

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #7298 

## Description of Changes
- adjusted breadcrumbs for contest pages
- adjusted mocked data to the real schema merged recently in #7518 
- adjusted ManageContestForm to be able to handle both creation and edition of the contest:
  - created temporary state object
  - disabled input fields
- extracted topic-related logic to its own hook 

## Test Plan
- you need to be admin of community that have stake enabled + have `FLAG_CONTEST=true` in `.env`
- everything is still based on mocked data
- you should be able to go through the launch contest flow (contest won't be created on save/sign but it looks like it is) - go to`communityId/manage/contests/launch`
- you can edit existing contests by going to `communityId/manage/contests` and clicking `edit contest` - save will not work as it is mocked data

https://github.com/hicommonwealth/commonwealth/assets/14819225/bb582797-2246-4e6a-a2e3-a8a16eef9deb

## Deployment Plan

1. still behind the feature flag

### CODE REVIEW

**Review by burtonator — APPROVED** [review]

File: packages/commonwealth/client/styles/components/component_kit/new_designs/cw_radio_button.scss [hunk 1]
```diff
@@ -27,14 +27,6 @@
...
       accent-color: $primary-600;
     }
-
-    &.disabled {
-      pointer-events: none;
-      box-shadow: inset 0 0 0 1px $neutral-200;
-      border-radius: 50%;
-      appearance: none;
-      background-color: $neutral-100;
-    }
```
Comment (mzparacha) [inline packages/commonwealth/client/styles/components/component_kit/new_designs/cw_radio_button.scss:37]: i think we shouldn't remove this as its used in https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/components/poll_card.tsx

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx [hunk 2]
```diff
@@ -111,63 +99,78 @@ const DetailsFormStep = ({
...
+          buttonHeight: 'sm',
+          onClick: goBack,
+        },
+      ],
+    });
   };
 
-  const handleSubmit = (values: ContestFormSubmitValues) => {
-    console.log('values', values);
+  const handleSubmit = (values: ContestFormValidationSubmitValues) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx:127]: i found that zod provides typescript types for subimt values, we can do
```
values: z.infer<typeof detailsFormValidationSchema>,
```

File: packages/commonwealth/client/scripts/views/components/share_popover.tsx [hunk 3]
```diff
@@ -72,14 +78,19 @@ export const SharePopover = ({
...
-            if (!commentId) {
-              await window.open(
-                `https://twitter.com/intent/tweet?text=${domain}${discussionLink}`,
+            if (discussionLink) {
+              window.open(
+                `${twitterPrefix}${domain}${discussionLink}`,
+                '_blank',
+              );
+            } else if (commentId) {
+              window.open(
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/share_popover.tsx:87]: Share on twitter seems to be broken

Uploading Screen Recording 2024-04-26 at 7.06.14 PM.mov…

File: packages/commonwealth/client/styles/components/component_kit/new_designs/cw_radio_button.scss (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/styles/components/component_kit/new_designs/cw_radio_button.scss:37]: 1. This file is radio button from `component_kit/new_designs` and one from `poll_card` uses old radio button from `component_kit/cw_radio_button` so this is something different
2. I removed it as it did not work properly - when the value was selected but the state was disabled, there was no visual affordance that the radio is selected. Now it's look like this which is good 
image

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx:127]: this is exactly what I do but just exported as a type in [this file](https://github.com/hicommonwealth/commonwealth/blob/f979a0dd795f842eb6cf16f55f29ed82ab509387/packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/types.ts#L19)

image

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx:127]: oh nice. Didn't notice that. Thanks for explaination

File: packages/commonwealth/client/scripts/views/components/share_popover.tsx [hunk 4]
```diff
@@ -84,8 +84,10 @@ export const SharePopover = ({
                 '_blank',
               );
             } else if (commentId) {
+              const currentRouteSansCommentParam =
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/share_popover.tsx:87]:
```diff
- currentRouteSansCommentParam
+ currentRouteHasCommentParam
```

**Review by mzparacha — APPROVED** [review]
Thanks for the changes, works great.

File: packages/commonwealth/client/scripts/views/components/share_popover.tsx (hunk 4 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/share_popover.tsx:87]: this actually should be `sans` as it is without comment param

**Pull request merged**